From f220494e338082d89c9c15c35bff0e2dea74f851 Mon Sep 17 00:00:00 2001
From: ChonDoit <thphantomblog@gmail.com>
Date: Fri, 30 Sep 2022 00:46:55 -0300
Subject: [PATCH] Automatically detect pick up sensor

So that an overlay is required for the sole purpose
of enabling pulse doze on pick up sensor

---
 .../display/AmbientDisplayConfiguration.java    | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/core/java/android/hardware/display/AmbientDisplayConfiguration.java b/core/java/android/hardware/display/AmbientDisplayConfiguration.java
index 7d8f2ff92200..500aa0a51193 100644
--- a/core/java/android/hardware/display/AmbientDisplayConfiguration.java
+++ b/core/java/android/hardware/display/AmbientDisplayConfiguration.java
@@ -22,6 +22,8 @@ import android.os.Build;
 import android.os.SystemProperties;
 import android.provider.Settings;
 import android.text.TextUtils;
+import android.hardware.SensorManager;
+import android.hardware.Sensor;
 import android.util.ArrayMap;
 import android.util.SparseArray;
 
@@ -100,7 +102,20 @@ public class AmbientDisplayConfiguration {
 
     /** @hide */
     public boolean dozePickupSensorAvailable() {
-        return mContext.getResources().getBoolean(R.bool.config_dozePulsePickup);
+        SensorManager sm = mContext.getSystemService(SensorManager.class);
+        boolean found = false;
+        if(sm == null) {
+            android.util.Log.d("PHH", "Failed getting sensor manager, can't detect pickup sensor");
+        } else {
+            java.util.List<Sensor> sensors = sm.getSensorList(Sensor.TYPE_ALL);
+            for(Sensor s : sensors) {
+                if(Sensor.STRING_TYPE_PICK_UP_GESTURE.equals(s.getStringType())) {
+                    found = true;
+                    break;
+                }
+            }
+        }
+        return mContext.getResources().getBoolean(R.bool.config_dozePulsePickup) || found;
     }
 
     /** @hide */
-- 
2.37.3

